### DISCUSSION TITLE
How to impove sub rate

### DISCUSSION THREAD

**Discussion by jiajiale** [emqx/emqx#4610] [Q&A] [answered]
Environment
  OS: Centos 8.0
  EMQ: 4.2.9
  CPU: 2vcpu
  Memory: 8G
Description
I change listener.tcp.external.max_conn_rate parameter to 10000，but when I subscribe emqx, only get 1k sub rate, Is there any other parameters I need to change?

**Comment by zmstone** [comment, marked as answer]
It's the same questions as https://github.com/emqx/emqx/discussions/4654 (asked in Chinese).

We have some improvements to be released in 4.3 which improves subscribe rate in below scenarios
1. clustered
2. wildcard topics in use
3. non-wildcard parts of the topic prefixes are not shared between concurrent subscribe requests 
  for example `a/b/<uuid>/+/....` if the `<uuid>` part is before any of the wildcard level (`+` in this case)
  then there could be a significant improvement in subscription rate
